apps/docs/content/guides/auth/quickstarts/react.mdx

```diff
@@ -69,58 +94,193 @@ hideToc: true
+          if (authError) {
+              return (
+                  <div>
+                      <h1>Authentication</h1>
+                      <p>✗ Authentication failed</p>
+                      <p>{authError}</p>
+                      <button
+                          onClick={() => {
+                              setAuthError(null);
+                              window.history.replaceState({}, document.title, "/");
+                          }}
+                      >
+                          Return to login
+                      </button>
+                  </div>
+              );
           }
-          else {
-            return (<div>Logged in!</div>)
+
+          // Show auth success (briefly before session loads)
+          if (authSuccess && !session) {
+              return (
+                  <div>
+                      <h1>Authentication</h1>
+                      <p>✓ Authentication successful!</p>
+                      <p>Loading your account...</p>
+                  </div>
+              );
           }
-        }
+
+          // If user is logged in, show welcome screen
+          if (session) {
+              return (
+                  <div>
+                      <h1>Welcome!</h1>
+                      <p>You are logged in as: {session.user.email}</p>
+                      <button onClick={handleLogout}>
+                          Sign Out
+                      </button>
+                  </div>
+              );
+          }
+
+          // Show login form
+          return (
+              <div>
+                  <h1>Supabase + React</h1>
+                  <p>Sign in via magic link with your email below</p>
+                  <form onSubmit={handleLogin}>
+                      <input
+                          type="email"
+                          placeholder="Your email"
+                          value={email}
+                          required={true}
+                          onChange={(e) => setEmail(e.target.value)}
+                      />
+                      <button disabled={loading}>
+                          {loading ? <span>Loading</span> : <span>Send magic link</span>}
+                      </button>
+                  </form>
+              </div>
+          );
+      }
       ```
 
     </StepHikeCompact.Code>
 
   </StepHikeCompact.Step>
 
-  <StepHikeCompact.Step step={5}>
-    <StepHikeCompact.Details title="Start the app">
+  <StepHikeCompact.Step step={6}>
+    <StepHikeCompact.Details title="Customize email template">
+
+    Before proceeding, change the email template to support support a server-side authentication flow that sends a token hash:
 
-    Start the app, go to http://localhost:3000 in a browser, and open the browser console and you should be able to log in.
+    - Go to the Auth templates page in your dashboard.
+    - Select the Confirm signup template.
+    - Change `{{ .ConfirmationURL }}` to `{{ .SiteURL }}?token_hash={{ .TokenHash }}&type=email`.
```

Comment:
Vite points to localhost:5173 by default, so the user either needs to update their Supabase configs or update their site URL. This suggestion is for the latter approach
```suggestion
    - Change `{{ .ConfirmationURL }}` to `{{ .SiteURL }}?token_hash={{ .TokenHash }}&type=email`.
    - Change your [Site URL](https://supabase.com/dashboard/project/_/auth/url-configuration) to `https://localhost:5173`
```